cpp/include/cuml/manifold/umap.hpp

```diff
@@ -72,10 +80,15 @@ class UMAP_API {
    *        n_samples in X
    * @param d
    *        d_features in X
+   * @param knn_indices
+   *        an array containing the n_neighbors nearest neighors indices for each sample
+   * @param knn_dists
+   *        an array containing the n_neighbors nearest neighors distances for each sample
    * @param embeddings
    *        an array to return the output embeddings of size (n_samples, n_components)
    */
-  void fit(float *X, float *y, int n, int d, float *embeddings);
+  void fit(float *X, float *y, int n, int d, long *knn_indices,
```

Comment:
This gets a little more complicated in the supervised learning case, because if the expected `target_metric` is not discrete (eg. the labels are continuous values), we also need to perform (and accept into this function) a KNN on the labels. For the sake of time, however, I think we should create an issue for this and reference it in a TODO (perhaps in the cython).

Reply:
I believe this still needs to be done.